include/openssl/cmp.h

```diff
@@ -0,0 +1,238 @@
+/*-
+ * Copyright 2007-2018 The OpenSSL Project Authors. All Rights Reserved.
+ * Copyright Nokia 2007-2018
+ * Copyright Siemens AG 2015-2018
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ *
+ * CMP (RFC 4210) implementation by M. Peylo, M. Viljanen, and D. von Oheimb.
+ */
+
+#ifndef OSSL_HEADER_CMP_H
+# define OSSL_HEADER_CMP_H
+
+# include <openssl/opensslconf.h>
+
+# ifndef OPENSSL_NO_CMP
+#  include <openssl/crmf.h>
+#  include <openssl/cmperr.h>
+#  define OPENSSL_CMP_CONST const
```

Comment:
What is the purpose of this?

Reply:
This is used for standalone builds with older OpenSSL versions, where cmp.h gets patched as follows:
```
#  if OPENSSL_VERSION_NUMBER >= 0x10100007L
#  define OPENSSL_CMP_CONST const
#  else
#   define OPENSSL_CMP_CONST
#  endif
```

Reply:
Right - but as this isn't a standalone build any more can we lose it?

Reply:
Yep, we'll get rid of the `OPENSSL_CMP_CONST` here.

Reply:
Done. Replaced all `OPENSSL_CMP_CONST` by `const`.